doc(Fees): Add doc for fee advance in invoice (#170)
getlago/lago-doc: guide/plans/charges/invoiceable-vs-noninvoiceable.mdx

---
title: "Invoiceable vs Uninvoiceable"
description: "In-advance charges can be either invoiceable or not."
---

<Info>
  **Premium feature ✨**: only users with a premium license can define whether
  or not a charge is invoiceable. Please **[contact
  us](mailto:hello@getlago.com)** to get access to Lago Cloud and Lago
  Self-Hosted Premium.
</Info>

If a charge needs to be `paid in advance`, you can choose whether or not it generates an invoice.

The `"invoiceable": false` option is **particularly relevant for fintech companies that want to charge customers instantly
without issuing an invoice**. This feature can be useful for banking transactions like ATM withdrawals, FX transfers and other
similar scenarios.

> By setting the invoiceable parameter to false, a fee is created by default, and a webhook `fee.created` is sent immediately upon any changes in usage without generating an invoice. However, you can decide the charge invoicing strategy:

Reviewer: ```suggestion
By setting the invoiceable parameter to false, a fee is created by default, and a webhook `fee.created` is sent immediately upon any changes in usage without generating an invoice. However, you can configure the charge invoicing strategy via `regroup_paid_fees`:
```

1. `null`: Fees generated never appear on invoices (default behaviour).

> 2. `in_arrears`: Groups all paid fees into one invoice at the end of the billing period.

Reviewer: ```suggestion
2. `invoice`: Groups all paid fees into one invoice at the end of the billing period.
```

> Please note that if `in_arrears`, all fees generated with a `payment_status` different from `succeeded` won’t appear on invoice generated at the end of the billing period. They will appear on next ones if you manually edit their `payment_status` to `succeeded`.

Reviewer: ```suggestion
Please note that if `invoice`, all fees generated with a `payment_status` different from `succeeded` won’t appear on invoice generated at the end of the billing period. They will appear on next ones if you manually edit their `payment_status` to `succeeded`.
```
